### PR TITLE
feat: Celery tasks refactor

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We're trying to improve Celery and one part of this is code-base wise making it clearer and easier to manage.

## Changes

* Refactors all tasks to use `shared_task` to reduce the circuclar import issue
* Swaps the scheduled tasks to `on_after_finalize` as [mentioned here](https://docs.celeryq.dev/en/stable/userguide/periodic-tasks.html#entries)
* Moves almost all tasks into `tasks`
* Adds enum for choosing the queue

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Checked the worker starts up as expected. Will deploy a PR version to test there first.

### CODE REVIEW

File: posthog/celery.py [hunk 1]
```diff
@@ -382,715 +148,8 @@ def retry_signal_handler(sender, **kwargs):
...
-@app.task(ignore_result=True)
-def ee_persist_finished_recordings():
-    try:
-        from ee.session_recordings.persistence_tasks import persist_finished_recordings
-    except ImportError:
-        pass
-    else:
-        persist_finished_recordings()
-
+@app.on_after_finalize.connect
```
Comment (benjackwhite) [inline posthog/celery.py:151]: The big win here is now we don't have to ever worry about imports!

**Review by pauldambra — APPROVED** [review]
Celery starts and tasks run 🚢

File: posthog/celery.py (hunk 1 above)
Comment (pauldambra) [reply posthog/celery.py:151]: I'm certain that's a source of unknown bugs so that's massive

File: posthog/tasks/utils.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from enum import Enum
+
+
+# NOTE: Keep in sync with bin/celery-queues.env
+class CeleryQueue(Enum):
```
Comment (webjunkie) [inline posthog/tasks/utils.py:27]: I added a `stats` queue for the priority consumer as well. So we can add that here or later.

File: posthog/tasks/tasks.py [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+
+    ExportedAsset.delete_expired_assets()
+
+
+@shared_task(ignore_result=True)
+def redis_heartbeat():
+    get_client().set("POSTHOG_HEARTBEAT", int(time.time()))
+
+
+@shared_task(ignore_result=True, queue=CeleryQueue.ANALYTICS_QUERIES)
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog/tasks/tasks.py:30]: Yeah I'm fixing this atm

File: posthog/tasks/utils.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/tasks/utils.py]: I'll add it here 👍

**Pull request merged**